pkg/scheduler/core/generic_scheduler.go

```diff
@@ -45,28 +45,52 @@ func NewGenericScheduler(
 	}
 }
 
-func (g *genericScheduler) Schedule(ctx context.Context, placement *policyv1alpha1.Placement, spec *workv1alpha2.ResourceBindingSpec) (result ScheduleResult, err error) {
+// evict indicates whether to remove reserved clusters that don't fit the placement
+func (g *genericScheduler) Schedule(ctx context.Context, placement *policyv1alpha1.Placement, spec *workv1alpha2.ResourceBindingSpec, evict bool) (result ScheduleResult, err error) {
 	clusterInfoSnapshot := g.schedulerCache.Snapshot()
 	if clusterInfoSnapshot.NumOfClusters() == 0 {
 		return result, fmt.Errorf("no clusters available to schedule")
 	}
 
-	feasibleClusters, err := g.findClustersThatFit(ctx, g.scheduleFramework, placement, &spec.Resource, clusterInfoSnapshot)
+	// 1. calculate candidate clusters and reserved clusters
+	// 2. prioritize candidate clusters
+	// 3. select target clusters from (reserved clusters + candidate clusters)
+	// 4. assign replicas
+	readyClusters, unreadyClusters := clusterInfoSnapshot.InspectClusters()
+	bindClusters := util.ConvertToClusterNames(spec.Clusters)
+
+	// available clusters are ready clusters NOT in `binding.spec.clusters`
+	availableClusters := readyClusters.Difference(bindClusters)
+	// candidate clusters are available clusters that fit the placement
+	candidateClusters, err := g.findClustersThatFit(ctx, g.scheduleFramework, placement, &spec.Resource, clusterInfoSnapshot, availableClusters)
```

Comment:
> cluster A add a taint which the deployment doesn't tolerate
the replicas scale up from 5 to 10
new pod should not be scheduled to cluster A

We had discussed about this and we think the original scheduled clusters should be evicted **only when the propagation policy changes**

Reply:
Yeah it sounds reasonable, but we can not remove cluster A from `binding.spec.clusters` directly cos it will break the original workloads in cluster A

It's kind of tricky and IMHO it's something about `assignReplicas`, we may implement this in a follow-up